specs/error_state/ErrorOutOfGasSloadSstore.md

```diff
@@ -0,0 +1,62 @@
+# ErrorOutOfGasSloadSstore state for both SLOAD and SSTORE OOG errors
+
+## Procedure
+
+Handle the corresponding out of gas errors for both `SLOAD` and `SSTORE` opcodes.
+
+### EVM behavior
+
+For the current `go-ethereum` code, the out of gas error may occur for `constant gas` or `dynamic gas`.
+
+#### SLOAD gas cost
+
+For this gadget, SLOAD gas cost is calculated with EIP-2929 as:
+```
+if is_warm_access:
+  gas_cost = GAS_COST_WARM_ACCESS # 100
+else:
+  gas_cost = COLD_SLOAD_COST # 2100
+```
+
+#### SSTORE gas cost
+
+For this gadget, SSTORE gas cost is calculated with EIP-3529 as:
+```
+if value == value_prev:
+  gas_cost = SLOAD_GAS # 100
+else:
+  if value == original_value:
```

Comment:
Here should be `!=`. If it's a `==` which means `original_value = value == 0 & value != value_prev` ==> e.g. 0 --> non-zero --> 0. In this case, gas_cost should be `SLOAD_GAS` only.

Reply:
I think to use `original_eq_prev` [here](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1147/files#diff-08d58ebc59ab6c11bf7c049e859441eed2319a03feb3fdffaddaba73f8a8c829R719-R724) is incorrect. 

Let's see the case, `value_eq_prev == false && original_eq_prev == false`, the `gas_cost` would be `WARM_ACCESS`. The case means `original_value != value_pre != value` but it doesn't imply `original_value == value` and I can't find any contraints for this equivalence (correct me if I missed it).

So, I think to follow the spec here (using `value != original_value`) is correct.